PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/webapp/WebAppConfigFormPanelAdvanced.java

```diff
@@ -91,25 +108,27 @@ public void setVisible(final boolean visible) {
     private void init() {
         this.textSku.setBorder(new EmptyBorder(0, 5, 0, 0));
         this.textSku.setText(NOT_APPLICABLE);
-        this.selectorServicePlan.addItemListener(e -> {
-            if (e.getStateChange() == ItemEvent.SELECTED) {
-                final AppServicePlan plan = (AppServicePlan) e.getItem();
-                this.textSku.setText(plan.pricingTier().toString());
-            } else if (e.getStateChange() == ItemEvent.DESELECTED) {
-                this.textSku.setText(NOT_APPLICABLE);
-            }
-        });
-        this.selectorSubscription.addItemListener(e -> {
-            if (e.getStateChange() == ItemEvent.SELECTED) {
-                final Subscription subscription = (Subscription) e.getItem();
-                this.selectorGroup.refreshWith(subscription);
-                this.selectorServicePlan.refreshWith(subscription);
-                this.selectorRegion.refreshWith(subscription);
-            } else if (e.getStateChange() == ItemEvent.DESELECTED) {
-                this.selectorGroup.clear();
-                this.selectorServicePlan.clear();
-                this.selectorRegion.clear();
-            }
-        });
+        this.selectorServicePlan.addItemListener(this::onServicePlanChanged);
+        this.selectorSubscription.addItemListener(this::onSubscriptionChanged);
+    }
+
+    private void onSubscriptionChanged(final ItemEvent e) {
+        //TODO: @wangmi try subscription mechanism? e.g. this.selectorGroup.subscribe(this.selectSubscription)
+        if (e.getStateChange() == ItemEvent.SELECTED || e.getStateChange() == ItemEvent.DESELECTED) {
```

Comment:
If we change the subscription, will we get an DESELECTED event?

Reply:
no. 
this is to handle `ItemEvent.DESELECTED` event, I guess this event will be fired when user cancel select a item.
for this case, I need to tell the denpending component that subscription is cleared, you have to refresh your items and revalidate.